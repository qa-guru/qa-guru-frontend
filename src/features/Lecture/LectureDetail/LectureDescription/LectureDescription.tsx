import React from "react";
import { Divider, Paper, Stack, Typography } from "@mui/material";
import { primary, white } from "../../../../theme/colors";
import { ILectureDescription } from "./LectureDescription.types";

const style = {
  paper: { padding: "20px" },
  circle: {
    minWidth: "40px",
    height: "40px",
    bgcolor: primary.main,
    borderRadius: "50%",
    display: "flex",
    justifyContent: "center",
    alignItems: "center",
    color: white.main,
  },
};

const LectureDescription: React.FC<ILectureDescription> = (props) => {
  const { description } = props;

  console.log(description);

  return (
    <>
      <Paper sx={style.paper}>
        <Typography mb="14px" variant="h6">
          Содержание урока
        </Typography>
        <Divider />
        <Stack spacing={1.5}>
          {description?.map((value: any, index: any) => {
            return (
              <Stack
                direction="row"
                spacing={1.5}
                alignItems="center"
                key={index}
                mt="16px"
              >
                <Typography sx={style.circle} variant="subtitle2">
                  {index + 1}
                </Typography>
                <Typography variant="subtitle1">{value}</Typography>
              </Stack>
            );
          })}
        </Stack>
      </Paper>
    </>
  );
};

export default LectureDescription;
