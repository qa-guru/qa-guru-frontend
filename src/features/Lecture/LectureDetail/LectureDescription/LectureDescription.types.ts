export interface ILectureDescription {
  description: any;
}
