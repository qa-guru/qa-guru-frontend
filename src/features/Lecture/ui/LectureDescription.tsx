import React from "react";
import { Divider, Paper, Stack, Typography } from "@mui/material";

interface ILectureDescription {
  arrayDescription: string[];
}

const LectureDescription: React.FC<ILectureDescription> = (props) => {
  const { arrayDescription } = props;

  return (
    <>
      <Paper sx={{ padding: "20px" }}>
        <Typography mb="14px" variant="h4">
          Содержание урока
        </Typography>
        <Divider />
        <Stack spacing={1.5}>
          {arrayDescription?.map((item, index) => {
            return (
              <Typography key={index} variant="subtitle2" mt="20px">
                {item}
              </Typography>
            );
          })}
        </Stack>
      </Paper>
    </>
  );
};

export default LectureDescription;
