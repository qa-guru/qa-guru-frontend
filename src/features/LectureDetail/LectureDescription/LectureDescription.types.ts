export interface ILectureDescription {
  description: Array<string | null> | null;
}
