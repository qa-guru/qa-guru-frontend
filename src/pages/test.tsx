import { FC } from "react";
import { useParams } from "react-router-dom";

import TestContainer from "features/test/containers/test-container";

const TestPage: FC = () => {
  const { testId, trainingId, lectureId } = useParams<{
    testId: string;
    trainingId: string;
    lectureId: string;
  }>();

  // Проверяем, что все параметры присутствуют
  if (!testId || !trainingId || !lectureId) {
    return <div>Ошибка: отсутствуют необходимые параметры</div>;
  }

  return (
    <TestContainer
      testId={testId}
      trainingId={trainingId}
      lectureId={lectureId}
    />
  );
};

export default TestPage;
