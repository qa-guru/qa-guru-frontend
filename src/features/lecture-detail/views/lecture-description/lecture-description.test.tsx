import { MemoryRouter } from "react-router-dom";
import LectureDescription from "./lecture-description";
import { render } from "../../../../test/utils-test";

describe("LectureDescription", () => {
  it("the component is render", () => {
    const { asFragment } = render(
      <MemoryRouter>
        <LectureDescription description={["string, string", "string"]} />
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
