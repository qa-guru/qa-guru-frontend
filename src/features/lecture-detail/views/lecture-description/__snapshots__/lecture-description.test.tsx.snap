// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`LectureDescription > renders correctly 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-nxklxt-MuiPaper-root"
  >
    <h5
      class="MuiTypography-root MuiTypography-h5 css-8wzaua-MuiTypography-root"
    >
      Содержание урока
    </h5>
    <hr
      class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
    />
    <div
      class="MuiStack-root css-10kbc59-MuiStack-root"
    >
      <div
        class="MuiStack-root css-5gq4xp-MuiStack-root"
      >
        <h6
          class="MuiTypography-root MuiTypography-subtitle2 css-16en7dj-MuiTypography-root"
        >
          1
        </h6>
        <h6
          class="MuiTypography-root MuiTypography-subtitle1 css-10wpov9-MuiTypography-root"
        >
          string, string
        </h6>
      </div>
      <div
        class="MuiStack-root css-5gq4xp-MuiStack-root"
      >
        <h6
          class="MuiTypography-root MuiTypography-subtitle2 css-16en7dj-MuiTypography-root"
        >
          2
        </h6>
        <h6
          class="MuiTypography-root MuiTypography-subtitle1 css-10wpov9-MuiTypography-root"
        >
          string
        </h6>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
